CONTRIBUTING.md

```diff
@@ -1,122 +1,172 @@
+2. Follow the [MuleSoft Forum](http://forum.mulesoft.org/mulesoft) chatter to see if anyone else has started resolving the problem or initiated an improvement.
+3. Scan [StackOverflow](http://stackoverflow.com/questions/tagged/mule) to see if there is already a proposed solution to your problem.
 
-## Understanding the extension mechanisms
+### Understanding the Extension Mechanisms
 
-Mule Runtime has two different extension mechanisms for writing modules and connectors. Avoiding to add functionality to this project and rather use one of those mechanisms to extend Mule is probably the more favorable option.
+**Mule ESB** has two different extension mechanisms for writing modules and connectors.
+Avoiding adding functionality to the project. Rather, using one of the mechanisms to extend Mule is probably the more favorable option.
+It is important to understand both.
 
-It is therefor important to understand those two extension mechanisms. The first and more intuitive mechanism for extensions is the [Anypoint Connector DevKit](https://developer.mulesoft.com/docs/display/current/Anypoint+Connector+DevKit), an annotation based framework for easy construction of extensions. The second choice is to extend mule through the [Extensions](https://developer.mulesoft.com/docs/display/current/Extending) mechanism.
+- The **first** and more intuitive mechanism for extensions is the [Anypoint Connector DevKit](https://docs.mulesoft.com/anypoint-connector-devkit/v/3.7/index), an **annotation-based framework** for easy construction of extensions.
 
-Now we should understand Mule and the enterprise integration domain. At the same time we should  know the different extension mechanisms of Mule.
+- The **second** choice is to extend Mule through the [Extensions](https://docs.mulesoft.com/mule-fundamentals/v/3.7/extending-mule) mechanism.
 
- 
-# Setting up the development environment
+## Setting Up the Development Environment
+
+While getting ready to contribute to any piece of software, there are
+several prerequisite components you will have to install. Plus, you
+will need to obtain the preexisting source code there may exist.
+
+This section provides some guidelines for installing those components and downloading the source code.
+
+### Installing Prerequisites
+
+Before you get started, you need to set yourself up with an environment in which to develop Mule. Your **dev** environment needs the following elements:
 
-While getting ready to contribute to any piece of software we will need to install number of prerequisites, we will need also to obtain the preexisting source code there could be. In this section we will follow some installation steps for the prerequisites and also we will download the source code.
+* A **Java SDK**.
+* A recent version of **Maven**.
+* A development environment tool or *IDE* (most preferably Anypoint Studio with DevKit plugin installed).
```

Comment:
You don't need Anypoint Studio to contribute to Mule project, we should remove this note.